### PR TITLE
impl(generator): add client metadata to cancelled iam policy

### PR DESCRIPTION
**Pull request by alevenberg**
https://github.com/googleapis/google-cloud-cpp/issues/14107

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14130)

### CODE REVIEW

File: generator/integration_tests/golden/v1/golden_kitchen_sink_client.h [hunk 1]
```diff
@@ -21,6 +21,7 @@
 
 #include "generator/integration_tests/golden/v1/golden_kitchen_sink_connection.h"
 #include "google/cloud/future.h"
+#include "google/cloud/internal/make_status.h"
```
Comment (coryan) [inline generator/integration_tests/golden/v1/golden_kitchen_sink_client.h:24]: Seems like we get this `#include` whether it is used or not.

File: generator/internal/client_generator.cc [hunk 2]
```diff
@@ -78,8 +78,8 @@ Status ClientGenerator::GenerateHeader() {
                                   : vars("connection_rest_header_path"),
        IsExperimental() ? "google/cloud/experimental_tag.h" : "",
        "google/cloud/future.h", "google/cloud/options.h",
-       "google/cloud/polling_policy.h", "google/cloud/status_or.h",
-       "google/cloud/version.h"});
+       "google/cloud/polling_policy.h", "google/cloud/internal/make_status.h",
+       "google/cloud/status_or.h", "google/cloud/version.h"});
```
Comment (coryan) [inline generator/internal/client_generator.cc:82]: Maybe this should be:

```suggestion
       "google/cloud/polling_policy.h",
         (get_iam_policy_extension_ && set_iam_policy_extension_ ) ? 
 "google/cloud/internal/make_status.h" : "",
       "google/cloud/status_or.h", "google/cloud/version.h"});
```

Adding a helper to say `HasIamPolicyExtension()` would be cleaner too.

**Review by coryan — CHANGES_REQUESTED** [review]

File: generator/internal/client_generator.cc (hunk 2 above)
Comment (alevenberg) [reply generator/internal/client_generator.cc]: Good catch. Done!

File: generator/integration_tests/golden/v1/golden_kitchen_sink_client.h (hunk 1 above)
Comment (alevenberg) [reply generator/integration_tests/golden/v1/golden_kitchen_sink_client.h]: Removed if not used.

**Review by coryan — APPROVED** [review]

**Pull request merged**